app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetController.scala

```diff
@@ -120,10 +118,13 @@ class EditSetController @Inject() (
   def editRecord(id: String, recordId: String) = Action { implicit request: Request[AnyContent] =>
     withUser { _ =>
       logger.info(s"The edit set id is $id for record id $recordId")
+      val record = editSetRecords.getEditSetRecordByOCI(recordId).getOrElse(defaultEditSetRecord)
       val messages: Messages = request.messages
       val title: String = messages("edit-set.record.edit.title")
-      val heading: String = messages("edit-set.record.edit.heading")
-      Ok(editSetRecordEdit(title, heading, editSetRecordForm))
+      val heading: String = messages("edit-set.record.edit.heading", record.ccr)
+
+      val recordForm = editSetRecordForm.fill(record)
+      Ok(editSetRecordEdit(title, heading, recordForm))
```

Comment:
I think rather than returning the `defaultEditSetRecord` it would be better to return an HTTP 404 Not Found error. You could do this like so:-
```
editSetRecords.getEditSetRecordByOCI(recordId) match {
    case Some(record) =>
        val messages: Messages = request.messages
        val title: String = messages("edit-set.record.edit.title")
        val heading: String = messages("edit-set.record.edit.heading", record.ccr)
        val recordForm = editSetRecordForm.fill(record)
        Ok(editSetRecordEdit(title, heading, recordForm))
    case None => NotFound
}
```